src/main/java/net/dv8tion/jda/api/entities/ThreadChannel.java

```diff
@@ -71,119 +136,319 @@ default GuildMessageChannel getParentMessageChannel()
 
+    /**
+     * Gets a {@link ThreadMember} of this thread by their {@link Member}.
+     *
+     * @param member
+     *        The member to get the {@link ThreadMember} for.
+     *
+     * @return The {@link ThreadMember} of this thread for the given member.
+     */
+    //TODO-v5: docs - how much of this relies on the GUILD_MEMBERS intent. Current impl is unfinished and so cannot document.
     @Nullable
     default ThreadMember getThreadMember(Member member)
     {
         return getThreadMemberById(member.getId());
     }
 
+    //TODO-v5: docs - how much of this relies on the GUILD_MEMBERS intent. Current impl is unfinished and so cannot document.
     @Nullable
     default ThreadMember getThreadMember(User user)
     {
         return getThreadMemberById(user.getId());
     }
 
+    //TODO-v5: docs - how much of this relies on the GUILD_MEMBERS intent. Current impl is unfinished and so cannot document.
     @Nullable
     default ThreadMember getThreadMemberById(String id)
     {
         return getThreadMemberById(MiscUtil.parseSnowflake(id));
     }
 
+    //TODO-v5: docs - how much of this relies on the GUILD_MEMBERS intent. Current impl is unfinished and so cannot document.
     @Nullable
     ThreadMember getThreadMemberById(long id);
 
+    /**
+     * Retrieves the {@link ThreadMember ThreadMembers} of this thread.
+     *
+     * @return a RestAction that resolves into a List of {@link ThreadMember ThreadMembers} of this thread.
+     */
+    //TODO-v5: docs - documentation depends on implementation cleanup.
     @CheckReturnValue
     RestAction<List<ThreadMember>> retrieveThreadMembers();
 
+    /**
+     * Whether the current account is the owner of this thread.
+     *
+     * @return true if the self account is the owner of this thread, false otherwise.
+     */
     default boolean isOwner()
     {
         return getJDA().getSelfUser().getIdLong() == getOwnerIdLong();
     }
 
+    /**
+     * Gets the ID of the owner of this thread as a long.
+     *
+     * @return the ID of the member who created this thread as a long.
+     */
     long getOwnerIdLong();
 
+    /**
+     * Gets the {@link User} of the owner of this thread as a String.
+     *
+     * @return The {@link User} of the member who created this thread as a String.
+     */
     @Nullable
     default String getOwnerId()
     {
         return Long.toUnsignedString(getOwnerIdLong());
     }
 
+    /**
+     * Gets the {@link User} of the owner of this thread.
+     * <br>
+     * This will be null if the member is not cached
+     *
+     * @see #getThreadMemberById(long)
+     *
+     * @return The {@link User} of the member who created this thread.
+     */
     @Nullable
     default Member getOwner()
     {
         return getGuild().getMemberById(getOwnerIdLong());
     }
 
+    /**
+     * Gets the owner of this thread as a {@link ThreadMember}.
+     * <br>
+     * This will be null if the member is not cached.
```

Comment:
This will also be null if the member isn't in `getThreadMembers()`. i.e: A LOT

Definitely recommend `@see retrieveThreadMember`